components/stepper/index.css

```
@@ -10,134 +10,560 @@ OF ANY KIND, either express or implied. See the License for the specific languag
+    border-color: transparent;
+  }
 }
 
 .spectrum-Stepper-stepUp {
-  border-block-end: none;
-  border-end-end-radius: var(--spectrum-stepper-border-radius-reset);
+  padding-block-start: var(--mod-stepper-icon-nudge-start, var(--spectrum-stepper-icon-nudge-start));
 
-  padding-block-start: var(--spectrum-stepper-icon-nudge-top);
+  border-start-start-radius: var(--mod-stepper-button-border-radius, var(--spectrum-stepper-button-border-radius));
+  border-start-end-radius: var(--mod-stepper-button-border-radius, var(--spectrum-stepper-button-border-radius));
+  border-end-start-radius: 0;
+  border-end-end-radius: 0;
 }
 
 .spectrum-Stepper-stepDown {
-  /*border-block-start: none;*/
-  border-start-end-radius: var(--spectrum-stepper-border-radius-reset);
+  border-block-start-width: var(--mod-stepper-button-middle-border-width, var(--spectrum-stepper-button-middle-border-width));
+  padding-block-end: var(--mod-stepper-icon-nudge-end, var(--spectrum-stepper-icon-nudge-end));
 
-  padding-block-end: var(--spectrum-stepper-icon-nudge);
+  border-start-start-radius: 0;
+  border-start-end-radius: 0;
+  border-end-start-radius: var(--mod-stepper-button-border-radius, var(--spectrum-stepper-button-border-radius));
+  border-end-end-radius: var(--mod-stepper-button-border-radius, var(--spectrum-stepper-button-border-radius));
 }
 
 .spectrum-Stepper-textfield {
   flex: 1;
   inline-size: auto;
-}
-
-.spectrum-Stepper-input {
-  min-inline-size: 0;
-  border-start-end-radius: var(--spectrum-stepper-border-radius-reset);
-  border-end-end-radius: var(--spectrum-stepper-border-radius-reset);
-}
-
-.spectrum-Stepper-textfield {
   min-inline-size: 0;
-}
 
-.spectrum-Stepper--quiet {
-  border-radius: var(--spectrum-stepper-border-radius-reset);
-  inline-size: var(--spectrum-stepper-quiet-width);
-
-  .spectrum-Stepper-buttons {
-    border-radius: var(--spectrum-stepper-border-radius-reset);
+  /* remove textfield focus indicator so we don't have two of them */
+  &::after {
+    display: none;
   }
+}
 
-  .spectrum-Stepper-stepUp,
-  .spectrum-Stepper-stepDown {
-    inline-size: var(--spectrum-stepper-quiet-button-width);
-    min-inline-size: 0;
-
-    border-inline-end-width: var(--spectrum-stepper-border-size-reset);
-    border-block-start: none;
-    border-inline-start: none;
-    border-radius: var(--spectrum-stepper-border-radius-reset);
-
-    padding-inline-end: 0;
-    justify-content: flex-end;
-
-    /* More hitarea */
-    &:after {
-      content: '';
-      position: absolute;
-      block-size: 100%;
-      inline-size: var(--spectrum-stepper-button-offset);
-      inset-inline-end: calc(var(--spectrum-stepper-button-offset) * -1);
-    }
-  }
+/* Only remove input border if stepper is visible */
+.spectrum-Stepper:not(.hide-stepper) .spectrum-Stepper-input {
```

```suggestion
.spectrum-Stepper:not(hidden) .spectrum-Stepper-input {
```

Instead of adding a hidden class, should we leverage the hidden attribute?

---

I'd vote against the hidden attribute just because it's very weak and easily overridden. Any display property will override it so I would have to add 
`[hidden] { display: none !important; }` which I just don't like to use `!important` but I can use it if you'd prefer